src/components/InfiniteScrollPaginator/InfiniteScroll.tsx

```diff
@@ -10,43 +12,78 @@ const mousewheelListener = (event: Event) => {
   }
 };
 
-export type InfiniteScrollProps = {
+export type InfiniteScrollProps = PaginatorProps & {
   className?: string;
   element?: React.ElementType;
+  /**
+   * @desc Flag signalling whether more pages with older items can be loaded
+   * @deprecated Use hasPreviousPage prop instead. Will be removed with v11.0.0.
+   */
   hasMore?: boolean;
+  /**
+   * @desc Flag signalling whether more pages with newer items can be loaded
+   * @deprecated Use hasNextPage prop instead. Will be removed with v11.0.0.
+   */
   hasMoreNewer?: boolean;
-  /** Element to be rendered at the top of the thread message list. By default Message and ThreadStart components */
+  /** Element to be rendered at the top of the thread message list. By default, Message and ThreadStart components */
   head?: React.ReactNode;
   initialLoad?: boolean;
   isLoading?: boolean;
   listenToScroll?: (offset: number, reverseOffset: number, threshold: number) => void;
   loader?: React.ReactNode;
-  loading?: React.ReactNode;
+  /**
+   * @desc Function that loads previous page with older items
+   * @deprecated Use loadPreviousPage prop instead. Will be removed with v11.0.0.
+   */
   loadMore?: () => void;
+  /**
+   * @desc Function that loads next page with newer items
+   * @deprecated Use loadNextPage prop instead. Will be removed with v11.0.0.
+   */
   loadMoreNewer?: () => void;
   pageStart?: number;
-  threshold?: number;
   useCapture?: boolean;
 };
 
 export const InfiniteScroll = (props: PropsWithChildren<InfiniteScrollProps>) => {
   const {
     children,
     element = 'div',
-    hasMore = false,
-    hasMoreNewer = false,
+    hasMore,
+    hasMoreNewer,
+    hasNextPage,
+    hasPreviousPage,
     head,
     initialLoad = true,
-    isLoading = false,
+    isLoading,
     listenToScroll,
     loader,
+    LoadingIndicator,
     loadMore,
     loadMoreNewer,
+    loadNextPage,
+    loadPreviousPage,
     threshold = 250,
     useCapture = false,
     ...elementProps
   } = props;
 
+  deprecationAndReplacementWarning(
+    [
+      [{ hasMoreNewer }, { hasNextPage }],
+      [{ loadMoreNewer }, { loadNextPage }],
+      [{ hasMore }, { hasPreviousPage }],
+      [{ loadMore }, { loadPreviousPage }],
+      [{ loader }, { LoadingIndicator }],
+    ],
+    'InfiniteScroll',
+  );
```

Comment:
Does this emit deprecation warnings on every render? If that is the case, I think we should log the deprecation message only once in order to prevent flooding the console with redundant log statements.

Reply:
@oliverlaz I have wrapped it with useEffect